llama_index/embeddings/openai.py

```diff
@@ -205,42 +221,27 @@ def __init__(
     ) -> None:
         """Init params."""
         super().__init__(**kwargs)
-        self.mode = OpenAIEmbeddingMode(mode)
-        self.model = OpenAIEmbeddingModelType(model)
         self.deployment_name = deployment_name
+        self.query_engine = get_engine(mode, model, _QUERY_MODE_MODEL_DICT)
+        self.text_engine = get_engine(mode, model, _TEXT_MODE_MODEL_DICT)
 
     def _get_query_embedding(self, query: str) -> List[float]:
         """Get query embedding."""
-        if self.deployment_name is not None:
-            engine = self.deployment_name
-        else:
-            key = (self.mode, self.model)
-            if key not in _QUERY_MODE_MODEL_DICT:
-                raise ValueError(f"Invalid mode, model combination: {key}")
-            engine = _QUERY_MODE_MODEL_DICT[key]
-        return get_embedding(query, engine=engine)
+        return get_embedding(
+            query, engine=self.query_engine, deployment_id=self.deployment_name
```

Comment:
is it ok to specify both engine and deployment_id for azure?

Reply:
![image](https://github.com/jerryjliu/llama_index/assets/6737785/a8ddcd44-404a-416f-b83d-6e64c92c0ebc)